docs: redoing for what was pr 4672 (#4676)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.2/release-notes/2.2.0/index.md

### Minio Disk insufficient space when upgrading

> When upgrading DKP from v2.1.x to v2.2.x, the upgrade can fail due to insufficient space on the MinIO Disk. To avoid this issue, we recommend that you disable the `fluent-bit` Platform Application before upgrading.

Reviewer: ```suggestion
When upgrading DKP from v2.1.x to v2.2.x, the upgrade can fail due to insufficient space on the MinIO Disk. To avoid this issue, D2iQ recommends that you disable the `fluent-bit` Platform Application before upgrading.
```

> ### Calico not updated during DKP upgrade

Author: Should it get called out sooner that this is only on Flatcar?

```suggestion
### Calico not updated during DKP upgrade on Flatcar
```

> When upgrading a DKP cluster, after what seems to be a successful upgrade, the Calico service might not update as expected and, therefor, is still using the old image. The wrong CNI ClusterResourceSet is being generated and not accounting for Flatcar. This issue only impacts Calico, no other add-ons.

Author: From @cneth 

```suggestion
When upgrading a DKP cluster running on Flatcar OS, you may find that after the upgrade the Calico services were not updated. This occurs because the upgrade procedure is not correctly updating the Flatcar specific CNI ClusterResourceSet(CRS). This issue only impacts the Calico CRS.
```

Reviewer 2: @msdolbey @cmastr - we need to pick one of these two and then make sure they get carried over to the 2.3/2.4/2.5 pages:

https://d2iq.atlassian.net/wiki/spaces/DENT/pages/150831110/DKP+2.5.0+Known+Issues+and+Limitations

Reviewer: Gonna commit Matt's then make new changes. We will need a JIRA to track 2.5, or we'll need to copy paste it now before we merge this PR.

> When upgrading a DKP cluster, after what seems to be a successful upgrade, the Calico service might not update as expected and, therefor, is still using the old image. The wrong CNI ClusterResourceSet is being generated and not accounting for Flatcar. This issue only impacts Calico, no other add-ons.

Reviewer: ```suggestion
When upgrading a DKP cluster, after what seems to be a successful upgrade, the Calico service may not update as expected and, therefore, is still using an old image. The wrong CNI ClusterResourceSet is being generated and not accounting for Flatcar. This issue only impacts Calico, no other applications.
```

Follow these steps to manually correct this issue:

1.  Update the ConfigMap as follows:

    ```yaml
    cat <<EOF | kubectl apply -f -
    apiVersion: v1
    data:
      custom-resources.yaml: |+
        # This section includes base Calico installation configuration.
        # For more information, see: https://docs.projectcalico.org/reference/installation/api
        apiVersion: operator.tigera.io/v1
        kind: Installation
        metadata:
          name: default
        spec:
          # Configures Calico networking.
          calicoNetwork:
            # Note: The ipPools section cannot be modified post-install.
            ipPools:
            - blockSize: 26
              cidr: 192.168.0.0/16
              encapsulation: IPIP
              natOutgoing: Enabled
              nodeSelector: all()
            bgp: Enabled
            nodeAddressAutodetectionV4:
              firstFound: true
          # FlexVolume path must be mounted under /opt on flatcar/coreos systems
          flexVolumePath: /opt/libexec/kubernetes/kubelet-plugins/volume/exec/
    kind: ConfigMap
    metadata:
      name: calico-cni-installation-$CLUSTER_NAME
    EOF
    ```

> 1.  Execute these commands: `kubectl edit clusterresourceset calico-cni-installation-$CLUSTER_NAME` and update `spec.clusterSelector.matchLabels.konvoy.d2iq.io/osHint` to `konvoy.d2iq.io/osHint: flatcar`.

Author: @dkoshkin @cneth - I just committed the CLUSTER_NAME business and the clusterresourceset here

> 1.  Execute these commands: `kubectl edit clusterresourceset calico-cni-installation-$CLUSTER_NAME` and update `spec.clusterSelector.matchLabels.konvoy.d2iq.io/osHint` to `konvoy.d2iq.io/osHint: flatcar`.

Reviewer: ```suggestion
1.  Run these commands: 

`kubectl edit clusterresourceset calico-cni-installation-$CLUSTER_NAME` and update `spec.clusterSelector.matchLabels.konvoy.d2iq.io/osHint` to `konvoy.d2iq.io/osHint: flatcar`.
```
